### PR TITLE
handlers: First scan ops

### LINKED ISSUE #217: scanmem: assert triggered with `0> <operator> <number>`
hello. After installing and launching scanmem, I tryed to find the first value. Bellow are all steps I did (I had truncated the less important informations):

```
$ scanmem
scanmem version 0.13

Enter the pid of the process to search using the "pid" command.

0> pid 12810
info: maps file located at /proc/12810/maps opened.
info: 107 suitable regions found.
0> = 126
info: 01/107 searching 0x7f41b6500000 - 0x7f41ba200000.scanmem: scanroutines.c:82: scan_routine_INTEGER8_NOTCHANGED: Assertion `old_value' failed.
error: 
Killed by signal 6.
```

### PR DESCRIPTION
**Pull request by 12345ieee**
Some improvements for scan operators handling.

Fixes #217 .

### CODE REVIEW

File: handlers.h [hunk 1]
```diff
@@ -113,25 +113,28 @@ bool handler__dregion(globals_t * vars, char **argv, unsigned argc);
...
                 "If n is given, match values that are less than n.\n" \
-                "Otherwise all current matches that have not decreased since the last scan are discarded.(same as `-`)\n" \
+                "Otherwise match all values that have decreased. (same as `-`)\n" \
                 "You can use this in conjunction with `snapshot` if you never know its value."
-#define NOTCHANGED_LONGDOC "usage: =\n" \
-                "All current matches that have changed since the last scan are discarded.\n" \
+
+#define NOTCHANGED_LONGDOC "usage: = [n]\n" \
+                "If n is given, match values that are equal to n. (same as `n`)\n" \
+                "Otherwise match all values have not changed since the last scan.\n" \
```
Comment (sriemer) [inline handlers.h]: "that" is missing here. I'll correct that during review.

File: handlers.h [hunk 2]
```diff
@@ -113,25 +113,28 @@ bool handler__dregion(globals_t * vars, char **argv, unsigned argc);
...
                 "You can use this in conjunction with `snapshot` if you never know its value."
-#define CHANGED_LONGDOC "usage: !=\n" \
-                "All current matches that have not changed since the last scan are discarded.\n" \
+#define CHANGED_LONGDOC "usage: != [n]\n" \
+                "If n is given, match values that are different from n.\n" \
+                "Otherwise match all values have changed since the last scan.\n" \
```
Comment (sriemer) [inline handlers.h]: Same issue here with missing "that".

**Comment by sriemer** [comment]
Didn't know that this is actually ready to be merged. It is merged now. Thanks!

File: handlers.c [hunk 3]
```diff
@@ -800,7 +800,7 @@ bool handler__decinc(globals_t * vars, char **argv, unsigned argc)
     }
     else
     {
-        show_error("unrecogised match type seen at decinc handler.\n");
+        show_error("unrecognised match type seen at decinc handler.\n");
```
Comment (sriemer) [inline handlers.c]: I've changed this to AE with "z".

**Comment by 12345ieee** [comment]
Thanks @sriemer for fixing my English.

I was looking at the documentation change and noticed the manpage doesn't document the alternative syntaxes of `= [n]` and `!= [n]`, but only `=` and `!=`.

I'll add them this evening and add a commit here, unless you get there first.

**Comment by sriemer** [comment]
I'll do it. Have to release earlier than that.

**Pull request closed without merging**